metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/ParserRunnerImpl.java

```diff
@@ -137,11 +208,29 @@ private void initializeParsers(Supplier<ParserConfigurations> parserConfigSuppli
     }
   }
 
+  /**
+   * Post-processes parsed messages by:
+   * <ul>
+   *   <li>Applying field transformations defined in the sensor parser config</li>
+   *   <li>Filtering messages using the configured MessageFilter class</li>
+   *   <li>Validating messages using the MessageParser validate method</li>
+   * </ul>
+   * If a message is successfully processed a message is returned in a ProcessResult.  If a message fails
+   * validation, a MetronError object is created and returned in a ProcessResult.  If a message is
+   * filtered out an empty Optional is returned.
+   *
+   * @param sensorType Sensor type of the message
+   * @param message Message parsed by the MessageParser
+   * @param rawMessage Raw message including metadata
+   * @param parser MessageParser for the sensor type
+   * @param parserConfigurations Parser configurations
+   */
   @SuppressWarnings("unchecked")
-  protected Optional<ParserResult> processMessage(String sensorType, JSONObject message, RawMessage rawMessage,
+  protected Optional<ProcessResult> processMessage(String sensorType, JSONObject message, RawMessage rawMessage,
                                                   MessageParser<JSONObject> parser,
```

Comment:
Is parser used?

Reply:
ignore the thing about Multiline... I'm mixing up PR's

Reply:
Yes parser is used.  Line 248, `parser.validate`.